src/octoprint/timelapse.py

```diff
@@ -838,7 +839,40 @@ def _render(self):
 		with self.render_job_lock:
 			try:
 				self._notify_callback("start", output)
-				p = sarge.run(command_str, stdout=sarge.Capture(), stderr=sarge.Capture())
+				p = sarge.run(command_str, stdout=sarge.Capture(), stderr=sarge.Capture(), async=True)
+
+				self._logger.debug("Parsing ffmpeg output")
+
+				# Compile regex to make it faster
+				duration_regex = re.compile("Duration: (\d{2}):(\d{2}):(\d{2})\.\d{2}")
+				current_regex = re.compile("time=(\d{2}):(\d{2}):(\d{2})\.\d{2}")
+				duration_s = ""
+
+				# Sarge sometimes doesn't populate commands fast enough, so we check if the command exists first, then poll the thread
+				while len(p.commands) == 0 or p.commands[0].poll() is None:
```

Comment:
This whole sarge processing should best use the same pattern we use at other places where we fire up other processes with `async=True` and try to parse their output on stdout and stderr, see for example [here](https://github.com/foosel/OctoPrint/blob/c55c568bbddf5112facf6aa3a173d500936b6cbf/src/octoprint/util/commandline.py#L73-L128). While that stuff looks slightly convoluted, there are reasons for all of those steps. 

In fact, it would make a lot of sense to just utilize [`util.commandline.CommandlineCaller`](https://github.com/foosel/OctoPrint/blob/c55c568bbddf5112facf6aa3a173d500936b6cbf/src/octoprint/util/commandline.py#L56) here (with custom logger methods) since it already does all of that stuff for us.